edgelet/hsm-rs/src/crypto.rs

```diff
@@ -194,6 +194,30 @@ fn make_certification_props(props: &CertificateProperties) -> Result<CERT_PROPS_
             unsafe { cert_properties_destroy(handle) };
             ErrorKind::CertProps
         })?;
+
+    if props.san_entries.len() > 0 {
+        let result: Result<Vec<CString>, NulError> = props
+            .san_entries
+            .iter()
+            .map(|s: &String| CString::new(s.clone()))
+            .collect();
+
+        let result: Vec<CString> = result.ok().ok_or_else(|| {
```

Comment:
`Result::ok` converts a `Result` into an `Option` and `Option::ok_or_else` converts an `Option` into a `Result`. So here we are effectively converting a `Result` into an `Option` and then back into a `Result`. If this is being done to change the error type then `Result::map_err` might be a better fit. Maybe.

Reply:
map_err is perfect for this. Modified the code to use.